lib/modules/manager/gitlabci/extract.ts

```diff
@@ -16,7 +16,7 @@ const imageRe = regEx(
 );
 const nameRe = regEx(`^\\s*name:\\s+['"]?(?<depName>[^\\s'"]+)['"]?\\s*$`);
 const serviceRe = regEx(
-  `^\\s*-?\\s*(?:name:\\s+)?['"]?(?<depName>[^\\s'"]+[^:]$)['"]?\\s*$`
+  `^\\s*-?\\s*(?:name:\\s+)?['"]?(?<depName>[\\{\\}\\$\\w.\\d\\/\\@]+:[\\{\\}\\$\\w.\\d\\:\\@-]+)['"]?\\s*$`
```

Comment:
Why do we need such a complex regex?

I really would prefer to use yaml based extraction, which would be more safe.

Reply:
I think even this regex could fail for multiple services:
```yaml
job1:
  services:
    - name: mooseagency/postgresql:12.3-1@sha256:a5a65569456f221ee1f8a0b3b4e2d440eb5830772d9440c9b30b1dbfd454c778
      command:
        - something:thatIsNotAnImage
        - -something2
        - -something3
        - -something4
    - "mariadb:10.4.11"
    - postgres:11.7
    - redis:latest
    - name: "registry.example.com/myimage:latest"
    - myimage@sha256:0ecb2ad60
    - tomcat:7-jre8
```

NB: Here you can find the schema for the gitlab-ci.yml: https://gitlab.com/gitlab-org/gitlab/-/raw/master/app/assets/javascripts/editor/schema/ci.json

Reply:
@viceice @rarkins I will do more research on this and change the method that is based on regex parsing